Cython/Compiler/Nodes.py

```diff
@@ -4852,6 +4854,31 @@ def analyse_declarations(self, env):
         if self.doc_node:
             self.doc_node.analyse_target_declaration(cenv)
         self.body.analyse_declarations(cenv)
+        self.analyze_class_annotations(env)
+
+    def analyze_class_annotations(self, env):
+        annotations = []
+        stat_list_node = self.body
+        for stat_node in stat_list_node.stats:
```

Comment:
I wonder if this is enough, or if we still need to descend into some tree nodes here. Probably for things like
```
try:
    x: int = may_fail()
except:
    ...
```
Look at `declare_from_annotation()` in `ExprNodes.py`. It already does a part of the job of collecting the annotations in the symbol table (`env.entries`). I think what we should rather do is to make sure the annotation always ends up there, whether we use it internally for static typing or not, and then collect the annotation dict for class scopes from all `Entry`s in `env.entries` that have an annotation. I think that's the cleanest way to do this.

Reply:
Okay - so want to make sure I follow you correctly - 

1. Let's clean this method up to iterate over `env.entries` and just build the annotation dict from the entries in that collection which have an annotation.
2. Let's make sure `declare_from_annotation()` is collecting valid cases (e.g., don't ignore class body entries).

I believe my latest updates should reflect this!